### PR TITLE
Downloading wrong starter project when using `odo init`

### LINKED ISSUE #6195: odo init downloads wrong starter project
/kind bug
/priority high

```
▶ odo version
odo v3.0.0-rc2 (3281c997c)
```

Using java-vertx devfile, I selected `vertx-cache-example-redhat` but  `vertx-istio-circuit-breaker-booster` was downloaded 

```
▶ odo init
  __
 /  \__     Initializing a new component
 \__/  \    Files: No source code detected, a starter project will be created in the current directory
 /  \__/    odo version: v3.0.0-rc2
 \__/

Interactive mode enabled, please answer the following questions:
? Select language: java
? Select project type: Vert.x Java
 ✓  Downloading devfile "java-vertx" from registry "DefaultDevfileRegistry" [1s]
? Which starter project do you want to use? vertx-cache-example-redhat
? Enter component name: asdf
 ✓  Downloading starter project "vertx-istio-circuit-breaker-booster" [482ms]

Your new component 'asdf' is ready in the current directory.
To start editing your component, use 'odo dev' and open this folder in your favorite IDE.
Changes will be directly reflected on the cluster.

```

[asciicast](https://asciinema.org/a/XFqyUBTaaWafHXbdc3mi4oigA)

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this:**

/kind bug

**What does this PR do / why we need it:**
This PR fixes downloading of the wrong devfile-starter when using `odo init` in interactive mode

**Which issue(s) this PR fixes:**
Fixes #6195 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: pkg/init/backend/interactive.go [hunk 1]
```diff
@@ -90,11 +90,22 @@ loop:
...
+func (starterProject StarterProjectSorter) Less(i, j int) bool {
+	return starterProject[i].Name < starterProject[j].Name
+}
+
 func (o *InteractiveBackend) SelectStarterProject(devfile parser.DevfileObj, flags map[string]string) (*v1alpha2.StarterProject, error) {
 	starterProjects, err := devfile.Data.GetStarterProjects(parsercommon.DevfileOptions{})
 	if err != nil {
 		return nil, err
 	}
+	sort.Sort(StarterProjectSorter(starterProjects))
```
Comment (feloy) [inline pkg/init/backend/interactive.go:108]: You could use the simpler function `sort.Slice`, which needs only the equivalent of the `Less` method

**Review by feloy — CHANGES_REQUESTED** [review]

**Review by rm3l — CHANGES_REQUESTED** [review]
Could you add an integration test for this (given the High priority of this user-facing issue)? Based on the explicit repro steps provided in #6195.

**Review by rm3l — COMMENTED** [review]
/approve

**Comment by feloy** [comment]
/override OpenShift-Integration-tests/OpenShift-Integration-tests
Devfile registry error

/override ci/prow/v4.11-integration-e2e

**Pull request merged**